### PR TITLE
update to 1.0.4

### PR DESCRIPTION
**Pull request by ocefpaf**
Try to sync with https://github.com/AnacondaRecipes/c99-to-c89

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,30 +1,79 @@
...
 
 source:
-  fn: c99-to-c89-{{ version }}.zip  # [win]
-  url: https://github.com/libav/c99-to-c89/releases/download/release-1.0.3/c99-to-c89-{{ version }}.zip  # [win]
-  md5: d58bc9741f258841aa3196a1b4155e51
+  - url: https://github.com/AnacondaRecipes/c99-to-c89/archive/release-1.0.4.tar.gz
+    sha256: 1df2ec430d34e9034d162969fe1a12cee50fa67cacdcc60d7b57b6c45b576467
+  # We need a libclang. No one builds them statically for Windows unfortunately (though the
+  # libav binaries have been built this way). I rename the libclang.dll in these to avoid
+  # conflicts with conda-forge's clangdev (which we cannot use at present due to compat).
```
Comment (isuruf) [inline recipe/meta.yaml]: @mingwandroid, what were the issues with conda-forge's clangdev?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,30 +1,79 @@
...
+  # LLVM 6 has changed too much and is no longer compatible. It is at the ABI level, but
+  # not semantically (Assertion failed: n_tokens == 2, file convert.c, line 559).
+  - url: http://releases.llvm.org/3.9.1/LLVM-3.9.1-win64.exe  # [win64]
+    sha256: 7ea2b7bc0de6b96a6ce11e6cfece7b84a31fb0c86c977f42ee178cba41517606  # [win64]
 
 build:
   number: 0
-  skip: True  # [unix]
-  
+  skip: true  # [win and vc!=14]
```
Comment (isuruf) [inline recipe/meta.yaml]: Should this ignore the `vc` `run_exports`?

File: recipe/tests/check.c [hunk 3]
```diff
@@ -0,0 +1,174 @@
+///////////////////////////////////////////////////////////////////////////////
```
Comment (isuruf) [inline recipe/tests/check.c:1]: @ocefpaf, can we use these files from the source tarball?

Comment (ocefpaf) [reply recipe/tests/check.c:1]: Sure. I'll try that.

**Comment by mingwandroid** [comment]
clangdev is much too modern. I spent a while figuring out the latest official release that still worked OK. The semantics have changed considerably in the cursor / code walking stuff.

While we could rummage around in old clangdev packages trying to find one that will work, I'd rather not force old clangdevs into the build environment incase of conflicts so I elected to just bundle an official binary in a non-conflicting location instead.

File: recipe/meta.yaml (hunk 1 above)
Comment (mingwandroid) [reply recipe/meta.yaml]: Conda-forge's clangdev is much too modern. I spent a while figuring out the latest official release that still worked OK. The semantics have changed considerably in the cursor / code walking stuff (and I spent *waaaay* too long on this package to try to modernise that part of it, please feel free if you are interested!)

While we could rummage around in old clangdev packages trying to find one that will work, I'd rather not force old clangdevs into the build environment incase of conflicts so I elected to just bundle an official DLL in a non-conflicting location instead.

Comment (mingwandroid) [reply recipe/meta.yaml]: But .. why do you ask this? I wrote what I thought was a comprehensive explanation in the comment:

https://github.com/conda-forge/c99-to-c89-feedstock/pull/6/files#diff-e178b687b10a71a3348107ae3154e44cR13

.. did you miss that or did it not make sense or something?

Comment (isuruf) [reply recipe/meta.yaml]: I was thinking of using clangdev 3.9.1 from conda-forge and link statically. clangdev builds static libraries so you don't need it at runtime for c99-to-c89. I was just curious by your comment about not being compatible.

Comment (mingwandroid) [reply recipe/meta.yaml]: It could be worth a try to link it statically, yes.

**Comment by ocefpaf** [comment]
Not sure what is wrong there b/c I used this syntax before with proj :-/

```
Executing: conda build recipe --quiet
Warning: Treating unknown selector 'vc' as if it was False.
Warning: Treating unknown selector 'vc' as if it was False.
The source field should be a dict, not list in file C:\projects\c99-to-c89-feedstock\recipe\meta.yaml.
```

@msarahan Am I doing something wrong or did this change?

**Comment by msarahan** [comment]
It looks right to me.  Seems like this might be a bug from 3.12.0.  https://github.com/conda/conda-build/pull/3041/files#diff-c86934c27c2f25579540ddb447fa45bdL334 seems like a likely culprit, but it's not obvious to me what is going wrong.

**Comment by ocefpaf** [comment]
> It looks right to me. Seems like this might be a bug from 3.12.0. conda/conda-build/pull/3041/files#diff-c86934c27c2f25579540ddb447fa45bdL334 seems like a likely culprit, but it's not obvious to me what is going wrong.

Thanks for letting me know I'm not crazy. This has been a terrible morning with virtually everything on my stack (gdal) is broken due to changes in things that are out of my hands :-(

I'm reverting this to @mingwandroid's recipe and if this passes I'll merge it to unblock https://github.com/conda-forge/xz-feedstock/pull/17 and start rebuild the stack, again...

**Pull request merged**